### ISSUE TITLE
Linear gradients bugged with 53.x

### ISSUE THREAD

**Issue by ewuerger** [Kozea/WeasyPrint#1485]
Hi,

First of all: Thank you very much for this great library! 

We are using weasyprint to generate many specification documents in the railway market. Since **version 53** our b64 encoded images are looking very weird. It seems like _linearGradient_ elements are not recognized anymore and coordinates are wrong too. I hope the following images help you to debug it.

![image](https://user-images.githubusercontent.com/50786483/140095145-f305fc0a-9466-4fc0-8ddf-b497e1d4fec1.png)
![image](https://user-images.githubusercontent.com/50786483/140095165-ad08eef1-ed88-437d-b823-c410db795f92.png)
(size difference is just b/c of my snipping tool precision)
### **test.svg**
```svg
<?xml version="1.0" encoding="utf-8" ?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:ev="http://www.w3.org/2001/xml-events"
    xmlns:xlink="http://www.w3.org/1999/xlink" baseProfile="full" height="100" version="1.1" width="100">
    <defs>
        <style>
            .Example {
                fill: url("#.FillGradient");
                stroke: #4A4A97;
            }
        </style>
        <linearGradient id=".FillGradient" x1="0%" x2="0%" y1="0%" y2="100%">
            <stop offset="0" stop-color="#EF2929" />
            <stop offset="1" stop-color="#FFFFFF" />
        </linearGradient>
    </defs>
    <rect class="Example" height="100" width="100" x="0" y="0" />
    <text dominant-baseline="middle" text-anchor="start" x="20" y="15">
        <tspan x="15" xml:space="preserve" y="50">Example</tspan>
    </text>
</svg>
```
### **Showcase script**
```python
import base64
import jinja2

svg = pathlib.Path("test.svg").read_text()
b64 = base64.standard_b64encode(svg.encode("utf-8"))
b64_svg = "".join(("data:image/svg+xml;base64,", b64.decode("ascii")))
template = """
<!DOCTYPE html>
<html>
<style>
article {
    page: content;
}
@page content {
    size: A4 portrait;
    margin: 2cm 2cm 2cm 2.5cm;
}
figure>img {
    max-width: 100%;
    margin-left: auto;
}
</style>
<body>
<article>
<p>weasyprint 53</p>
<figure><img src="{{b64_svg}}"/></figure>
</article>
</body>
</html>
"""
template = jinja2.Environment().from_string(template).render(b64_svg=b64_svg)
html_template = weasyprint.HTML(string=template, base_url=".")
html_template.write_pdf(target=pathlib.Path("test.pdf"))
```

For now we need to stick to weasyprint<=53.

**Comment by liZe** [comment]
Hello, and thanks a lot for this very well documented issue.

> First of all: Thank you very much for this great library!

💜

> It seems like _linearGradient_ elements are not recognized anymore and coordinates are wrong too.

There were actually two bugs… The linear gradient was missing because of quotes not being handled correctly in `url(…)` values. The coordinates were wrong because of a dirty values used as font descent and ascent. The new values are still wrong (because they’re not the real values of the font), but at least they take care of the font size and should give suitable results in simple cases.

They’re fixed in the master branch, and will also be included in the next 53.4 version.

> We are using weasyprint to generate many specification documents in the railway market.

Glad to hear that WeasyPrint is useful for you! Don’t hesitate to send us a mail if you’d like to share your experience, we’d be really happy to learn more about these mysterious railway specifications 🚆🚆🚆 😃.

**Comment by liZe** [comment]
(Oh, and we definitely miss tests for SVG, it’s an issue we’re aware of and that’s scheduled for [version 55](https://github.com/Kozea/WeasyPrint/milestone/32).)
